src/status_im/contexts/communities/events.cljs

```diff
@@ -399,22 +399,24 @@
      [:json-rpc/call :schema.common/rpc-call]]]])
 
 (rf/reg-event-fx :communities/get-revealed-accounts-success
- (fn [{:keys [db]} [community-id on-success revealed-accounts-js]]
+ (fn [{:keys [db]} [community-id on-success latest-request-to-join-js]]
    (when-let [community (get-in db [:communities community-id])]
-     (let [revealed-accounts
-           (reduce
-            (fn [acc {:keys [address] :as revealed-account}]
-              (assoc acc address revealed-account))
-            {}
-            (data-store.communities/<-revealed-accounts-rpc revealed-accounts-js))
-
+     (let [request-to-join (js->clj latest-request-to-join-js :keywordize-keys true)
+           revealed-accounts (
+                              reduce
```

Comment:
Since you are touching this code @vitvly, one improvement is that we have a utility to index a sequence by any value in `utils.collection/index-by`. In your case here you would index by `:address`. There are many examples in the repo if you want to check it out.

Reply:
Thank you, used it, much simpler without explicit `reduce`.